### PR TITLE
Prow jobs support running integration tests with kind

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes to reviewers**:
1. `config-generator` adds configuration to Prow jobs that has `needs-dind: true` to allow creating `kind` clusters in Prow job pods
2. Add a Prow job for test-infra repo to run a smoke integration tests Prow job with `kind`

/cc

### CODE REVIEW

**Comment by chizhg** [comment]
/cc @chaodaiG 
FYI @mattmoor

File: tools/config-generator/main.go [hunk 1]
```diff
@@ -325,7 +323,9 @@ func addExtraEnvVarsToJob(envVars []string, data *baseProwJobTemplateData) {
 
 // setupDockerInDockerForJob enables docker-in-docker for the given job.
 func setupDockerInDockerForJob(data *baseProwJobTemplateData) {
-	addVolumeToJob(data, "/docker-graph", "docker-graph", false, "")
+	addVolumeToJob(data, "/docker-graph", "docker-graph", false, []string{"emptyDir: {}"})
```
Comment (chaodaiG) [inline tools/config-generator/main.go:326]: Can you add a comment here?

**Comment by chaodaiG** [comment]
Left a comment
/lgtm
/hold

File: tools/config-generator/main.go (hunk 1 above)
Comment (chizhg) [reply tools/config-generator/main.go:328]: Done.

**Pull request merged**